qiskit/qpy/type_keys.py

```diff
@@ -93,6 +101,8 @@ class Value(TypeKeyBase):
     INTEGER = b"i"
     FLOAT = b"f"
     COMPLEX = b"c"
+    CASE_DEFAULT = b"d"
```

Comment:
I can move it if you think there's something more sensible for me to put it - I wasn't sure.  It's unlikely that control flow itself will gain more keys that need adding here, but we will need an entire new subtree to handle extended classical expressions (Qiskit/rfcs#31) at some point in the future (that's completely separate to this).  It's likely they'll be able to have an entirely separate namespace, though.

I'm not sure what advantage there is to having these separated enums; there's cross-dependencies between them already, so it's insufficient to just make a key unique within a single enum.  In the general case, the keys have to be globally unique.  I found that when trying to use an `r` key in `Value`: it clashed with an `r` for the Python builtin `range` in `Container`.  I feel like it might be cleaner to unify these into a global enumeration and enforce complete uniqueness; once we've got the enum, there's no real need for the ASCII characters to have any mnemonic value, and since the allowed keys in any given place in the QPY format are often a mixture of a few enums, they're not offering much type safety at the moment.

Reply:
That's fair enough, yeah.  I'm not really familiar with the QPY code at all, so I'm happy to put it somewhere else if you'd prefer.

We currently don't have any plans to extend the control-flow handling, though I suppose at some very indeterminate time in the future there _might_ be the concept of a relocatable subroutine call.  That's probably a year+ away at least, though, so I wouldn't worry about it.